server/gitlab/api.go

```diff
@@ -412,3 +412,23 @@ func (g *gitlab) ResolveNamespaceAndProject(
 	}
 	return "", "", ErrNotFound
 }
+
+func (g *gitlab) TriggerNewBuildPipeline(user *UserInfo, repo string, refToCommit string) (*TriggerPipelineResponse, error) {
```

Comment:
Can this function just return `go-gitlab`'s `Pipeline` struct instead of converting to our own type? Then there is no need for `server/gitlab/types.go`

Reply:
fixed here 79db715b51441c46a443ddc9387452f1dd08b362